### DISCUSSION TITLE
Why custom action do not have elevated privilege？

### DISCUSSION THREAD

**Discussion by ad-chuliang** [wixtoolset/issues#8725] [Questions]
I want to do some custom action with c++ function. In some where, the custom action need privilege to do something. However, the action execute sequence after InstallFinalized，the msi has asked for elevated privilege, and the function still without privilege. So, how to solve it?

**Comment by bevanweiss** [comment]
> I want to do some custom action with c++ function. In some where, the custom action need privilege to do something. However, the action execute sequence after InstallFinalized，the msi has asked for elevated privilege, and the function still without privilege. So, how to solve it?

Show your authoring.
Show your logs.

Are you running the custom action as Impersonate="yes"?

**Reply by ad-chuliang** [reply]
Wix code:
copy external file
CustomAction Id="CopyExternalFileToInstallPath" BinaryRef="CustomAction" DllEntry="CopyExternalFileToInstallPath" /
InstallExecuteSequence
Custom Action="CopyExternalFileToInstallPath" After="InstallFinalize" Condition="NOT Installed"/
/InstallExecuteSequence

Reply:
I have tried Impersonate="yes". Then the InstallExecuteSequence of CopyExternalFileToInstallPath should change to " Before="InstallFinalize" ". In the C++ code, I use CopyFileW function, which in WinBase.h, to copy some files to install folder. However, the install location in disc c, so that the CopyFileW should have elevated privilege to copy.
If I use bootstrapper application, before show msi dialog, a window launched with administrator privileges will show up. Then, CopyFileW could work, even location in disc c.
When only msi that not bundle msi in bootstrapper application, the window launched with administrator privileges will show before progress dialog. So, I doubt that administrator privileges window show and function still do not work for the reason of time that privileges window show.
